disperser/apiserver/disperse_blob_v2.go

```diff
@@ -38,9 +38,19 @@ func (s *DispersalServerV2) DisperseBlob(ctx context.Context, req *pb.DisperseBl
 		return nil, err
 	}
 
-	// Check against payment meter to make sure there is quota remaining
-	if err := s.checkPaymentMeter(ctx, req, start); err != nil {
-		return nil, err
+	if s.useControllerMediatedPayments {
+		// Use the new controller-based payment system
+		err := s.controllerClient.AuthorizePayment(ctx, req.GetBlobHeader())
+		if err != nil {
+			// Pass through the structured error from the controller
+			return nil, err
+		}
+	} else {
+		// Use the legacy payment metering system
+		// Check against payment meter to make sure there is quota remaining
+		if err := s.checkPaymentMeter(ctx, req, start); err != nil {
+			return nil, err
+		}
```

Comment:
Once this feature is ready to ship, I suggest prioritizing the removal of the legacy pathway ASAP.